src/main/java/org/trigon/hardware/SignalThreadBase.java

```diff
@@ -7,21 +7,40 @@
 import java.util.concurrent.ArrayBlockingQueue;
 import java.util.concurrent.locks.ReentrantLock;
 
+/**
+ * A class that represents a base for a signal thread.
+ */
 public class SignalThreadBase extends Thread {
     public static final ReentrantLock SIGNALS_LOCK = new ReentrantLock();
     protected final Queue<Double> timestamps = new ArrayBlockingQueue<>(100);
     private final ThreadInputsAutoLogged threadInputs = new ThreadInputsAutoLogged();
     private final String name;
     protected double odometryFrequencyHertz = 50;
 
+    /**
+     * Creates a new SignalThreadBase.
+     *
+     * @param name The name of the thread
+     */
     public SignalThreadBase(String name) {
         this.name = name;
     }
 
+    /**
+     * Sets the odometry frequency in hertz.
+     * The odometry frequency determines how often the robot's position and motion data are updated.
+     * A higher frequency will result in more frequent updates, but may also demand more processing power.
+     * Only used for Spark motors.
+     *
+     * @param odometryFrequencyHertz The odometry frequency in hertz
+     */
     public void setOdometryFrequencyHertz(double odometryFrequencyHertz) {
```

Comment:
1. something here breaks convention
2. maybe this should be renamed to setThreadFrequencyHertz